crates/sui-faucet/src/faucet/mod.rs

```diff
@@ -44,6 +44,17 @@ pub enum BatchSendStatusType {
     DISCARDED,
 }
 
+pub struct AppState<F = Arc<SimpleFaucet>> {
+    pub faucet: F,
+    pub config: FaucetConfig,
+}
+
+impl AppState {
+    pub fn new(faucet: Arc<SimpleFaucet>, config: FaucetConfig) -> Self {
+        Self { faucet, config }
+    }
+}
```

Comment:
nit: seeing as the implementation is generic over the faucet type, it seems to make sense to make the constructor generic as well, but if this whole set-up is unnecessary, the other option is to remove the generic parameter (not sure where else this gets used!)

```suggestion
impl<F> AppState {
    pub fn new(faucet: F, config: FaucetConfig) -> Self {
        Self { faucet, config }
    }
}
```